### PR TITLE
18 atm1b data

### LINKED ISSUE #18: Add support for and test BLATM1B v1, ILATM1B v1 and v2
The reader code for atm1b should be capable of handling these datasets already, but we have only confirmed support for ILATM1B v1 (the e2e test uses this as an example).

Confirm these data products work as expected. At a minimum, the `DatasetShortName` type needs to be updated.

Dataset landing pages:

* [Pre-IceBridge ATM L1B Qfit Elevation and Return Strength, Version 1 (BLATM1B)](https://nsidc.org/data/blatm1b/versions/1)
* [IceBridge ATM L1B Qfit Elevation and Return Strength, Version 1 (ILATM1B)](https://nsidc.org/data/ilatm1b/versions/1)
* [IceBridge ATM L1B Elevation and Return Strength, Version 2 (ILATM1B)](https://nsidc.org/data/ilatm1b/versions/2)

### PR DESCRIPTION
**Pull request by trey-stafford**
Resolves #18

Adds support for ILATM1B v2 and BLATM1B v1.

Also adds some code to deal with ATM1B qfit files that lack a header. see https://github.com/nsidc/iceflow/issues/35. 

----
~📚 Documentation preview 📚: https://readthedocs-preview--31.org.readthedocs.build/en/31/~

https://iceflow--31.org.readthedocs.build/en/31/

^RTD link generation should be fixed with ~https://github.com/nsidc/iceflow/pull/32~ #33

### CODE REVIEW

File: src/nsidc/iceflow/itrf/__init__.py [hunk 1]
```diff
@@ -12,7 +12,10 @@ def check_itrf(itrf_str: str) -> bool:
     the 2-digit or 4-digit year (e.g., "ITRF93" or "ITRF2008").
 
     ITRFs prior to the 2000s conventionally used a 2-digit year (e.g.,
-    "ITRF88")."""
+    "ITRF88").
+
+    ITRF2020 is recognized as "ITRF20".
```
Comment (trey-stafford) [inline src/nsidc/iceflow/itrf/__init__.py:17]: This is maddening

File: tests/integration/test_e2e.py
Comment (trey-stafford) [inline tests/integration/test_e2e.py:1]: These tests currently cover ILATM1B v1 and v2 workflow.

TODO: add another test for BLATM1B v1 after ECS downtime is over.

File: tests/unit/test_atm1b_itrf_range.py
Comment (trey-stafford) [inline tests/unit/test_atm1b_itrf_range.py:1]: These tests copied from `valkyrie`.

File: src/nsidc/iceflow/data/atm1b.py [hunk 2]
```diff
@@ -286,42 +310,122 @@ def _qfit_file_header(filepath: Path) -> str:
...
-def extract_itrf(filepath: Path) -> str:
-    ext = filepath.suffix
-
-    if ext == ".qi":
-        header = _qfit_file_header(filepath)
-        results = re.finditer(r"itrf\d{2,4}", header)
-        itrfs = list({result.group() for result in results})
+# NOTE: See the docstring of `_infer_qfit_itrf` for more context about these
+# hard-coded ITRF date-ranges.
+# TODO: a typed-dict might be better here.
```
Comment (trey-stafford) [inline src/nsidc/iceflow/data/atm1b.py:315]: These hard-coded ranges and the code to lookup an ITRF based on a date come straight from `valkyrie`, with some minor modifications.

**Review by mfisher87 — APPROVED** [review]

**Pull request merged**